app/models/collection.rb

```diff
@@ -37,7 +37,7 @@ def self.create_record(concept_id, revision_id, data_format, collection_data, op
     Collection.transaction do
       collection = Collection.find_or_create_by(concept_id: concept_id)
       collection.update_attributes!(short_name: short_name)
-      new_record = Record.create(recordable: collection, revision_id: revision_id, format: data_format, daac: daac_from_concept_id(concept_id))
+      new_record = Record.create(recordable: collection, revision_id: revision_id, format: data_format, daac: daac_from_concept_id(concept_id), campaign: extract_campaign(collection_data))
```

Comment:
So you could do it here.   Can you override Record.create?  don't pass it a campaign, do something like this (pseudocode)
def create(... args ...) 
  record = super.create(... args ..)
  # record is not a specific record format instance.
  record.campaign = record[campaign_column]
  return record
end